### PR TITLE
PLAT-543: check that validated state equals to last dirty satet in st…

### PR DESCRIPTION
**Pull request by ivanshibitov**
Check that validated state equals to last dirty state in state report.
Skipped up to completing of https://insolar.atlassian.net/browse/PLAT-402

### CODE REVIEW

File: ledger-core/virtual/integration/statereport_test.go [hunk 1]
```diff
@@ -179,3 +183,69 @@ func TestVirtual_VStateReport_BadState_StateAlreadyExists(t *testing.T) {
...
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+	typedChecker.VStateReport.Set(func(report *payload.VStateReport) bool {
+		require.Equal(t, report.ProvidedContent.LatestDirtyState, content.LatestDirtyState)
+		require.Equal(t, report.ProvidedContent.LatestValidatedState, content.LatestDirtyState)
+		return true
+	})
+
+	wait := server.Journal.WaitStopOf(&handlers.SMVStateReport{}, 1)
+	server.IncrementPulseAndWaitIdle(ctx)
+	waitWithTimeout(wait)
```
Comment (sexyhamster) [inline ledger-core/virtual/integration/statereport_test.go:239]: whats wrong with testutils.WaitSignalsTimed?

File: ledger-core/virtual/integration/statereport_test.go [hunk 2]
```diff
@@ -179,3 +183,69 @@ func TestVirtual_VStateReport_BadState_StateAlreadyExists(t *testing.T) {
...
+		}
+
+		wait := server.Journal.WaitStopOf(&handlers.SMVStateReport{}, 1)
+		server.SendPayload(ctx, payload)
+		waitWithTimeout(wait)
+	}
+
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+	typedChecker.VStateReport.Set(func(report *payload.VStateReport) bool {
+		require.Equal(t, report.ProvidedContent.LatestDirtyState, content.LatestDirtyState)
```
Comment (sexyhamster) [inline ledger-core/virtual/integration/statereport_test.go:232]: you can actually not only make sure, that report equals required. You can set chan here to await after pulse change.
And you don't need resending StateReport, so please `return false`

File: ledger-core/virtual/integration/statereport_test.go (hunk 1 above)
Comment (ivanshibitov) [reply ledger-core/virtual/integration/statereport_test.go:239]: yes, I forget about it

File: ledger-core/virtual/integration/statereport_test.go (hunk 2 above)
Comment (ivanshibitov) [reply ledger-core/virtual/integration/statereport_test.go:233]: no problem

**Pull request merged**